aten/src/ATen/ParallelNativeTBB.cpp

```diff
@@ -0,0 +1,75 @@
+#if AT_PARALLEL_NATIVE_TBB
+#include <ATen/Parallel.h>
+
+#include <atomic>
+
+#include "tbb/tbb.h"
+
+#ifdef _OPENMP
+#include <omp.h>
+#endif
+
+#ifdef TH_BLAS_MKL
+#include <mkl.h>
+#endif
+
+namespace at {
+
+namespace internal {
+  tbb::task_scheduler_init tbb_init(
```

Comment:
This is the trickiest part. It must be called before *anything* calls into TBB. I'm not even sure how nicely this plays with other packages that might use TBB. This might be one of the main reasons we need a static build for TBB, so that we can be sure this is correct. I also don't know what happens when the system uses two versions of TBB at the same time, i.e. if they can play nicely with each other.

Reply:
As I understand, static build is not going to work with TBB used by PyTorch, MKL and MKL-DNN. As of now we compile MKL-DNN statically into PyTorch, so they could use statically compiled TBB, but it seems this is not going to work with dynamically loaded MKL